crates/rattler_conda_types/src/version/mod.rs

```diff
@@ -1289,17 +1306,90 @@ mod test {
     }
 
     #[test]
-    fn starts_with_matches_extra_component_against_zero() {
+    fn starts_with_extra_components() {
+        // For glob matching (e.g., "1.0.0_version*"), versions with extra
+        // components should match the prefix.
         let version = Version::from_str("1.0.0_version").unwrap();
-        assert!(!Version::from_str("1.0.0_version1")
+        // "1.0.0_version1" starts with "1.0.0_version" (extra component "1")
+        assert!(Version::from_str("1.0.0_version1")
             .unwrap()
             .starts_with(&version));
+        // "1.0.0_version0" starts with "1.0.0_version" (extra component "0")
         assert!(Version::from_str("1.0.0_version0")
             .unwrap()
             .starts_with(&version));
-        assert!(!Version::from_str("1.0.0_version0foo")
+        // "1.0.0_version0foo" starts with "1.0.0_version" (extra components "0", "foo")
+        assert!(Version::from_str("1.0.0_version0foo")
+            .unwrap()
+            .starts_with(&version));
+
+        // But different base components should NOT match
+        assert!(!Version::from_str("1.0.0_other")
+            .unwrap()
+            .starts_with(&version));
+        assert!(!Version::from_str("1.0.0_ver")
+            .unwrap()
+            .starts_with(&version));
+
+        // Different segment structure should NOT match (PR #1791)
+        // "1.0.0_version1" has segment [version, 1]
+        // "1.0.0_version_2" has segments [version], [2]
+        // This ensures "1.0.0_version_2*" does NOT match "1.0.0_version1"
+        let version_with_extra_segment = Version::from_str("1.0.0_version_2").unwrap();
+        assert!(!Version::from_str("1.0.0_version1")
+            .unwrap()
+            .starts_with(&version_with_extra_segment));
+
+        // Different component values should NOT match
+        let version1 = Version::from_str("1.0.0_version1").unwrap();
+        assert!(!Version::from_str("1.0.0_version0")
+            .unwrap()
+            .starts_with(&version1));
+
+        // Extra components after matching prefix should match
+        assert!(Version::from_str("1.0.0_version1a")
+            .unwrap()
+            .starts_with(&version1));
+        assert!(Version::from_str("1.0.0_version0a")
             .unwrap()
             .starts_with(&version));
+
+        // Extra components in INTERMEDIATE segments should NOT match
+        // "1.1c.1" has segment [1, c] where "1.1.1" has segment [1]
+        // This is a structure mismatch, not just extra components at the end
+        assert!(!Version::from_str("1.1c.1")
+            .unwrap()
+            .starts_with(&Version::from_str("1.1.1").unwrap()));
+        assert!(!Version::from_str("1.1c1.1")
+            .unwrap()
+            .starts_with(&Version::from_str("1.1c.1").unwrap()));
+
+        // BUT zero components in prefix are treated as "no component" (implicit zeros)
+        // So "1.1c.1" starts with "1.1c0.1" because c0 == c (trailing zero)
```

Comment:
This one is a bit surprising but I am fine with it. Did you double check vs. Conda?